hazelcast/src/main/java/com/hazelcast/internal/config/ConfigValidator.java

```diff
@@ -144,6 +146,20 @@ public static void checkEvictionConfig(EvictionPolicy evictionPolicy,
         }
     }
 
+    public static void checkMaxSizeEvictionConfig(MapConfig mapConfig, int partitionCount) {
```

Comment:
I think this is the first time the `ConfigChecker` changes a config. Maybe it's cleaner to return a `boolean` value with the check result and do the `maxSizeConfig.setSize(partitionCount)` at the caller side? Of course this would need a different test setup then.